### PR TITLE
feat(lazy): lazy.nvim settings can be customized

### LINKED ISSUE #3923: allow for customization to the lazy.nvim setup #3922
### Feature Description

There should be a way to add some customization to the lazy.nvim setup.
Starting with things like checker = { enabled = true, concurrency = 1 } to check for updates on startup.

### Describe the alternatives you have considered

The following would be possible:
```
local opts = {
  defaults = { lazy = true, version = "" },
  install = { missing = true, colorscheme = { "kanagawa" } },
  ui = {
    border = "rounded",
  },
 checker = {
    enabled = true,
    concurrency = 1,
 },
  performance = {
    cache = {
    enabled = true,
    },
    rtp = {
      disabled_plugins = {
        "gzip",
        "matchit",
        "matchparen",
        "netrwPlugin",
        "rplugin",
        "tarPlugin",
        "tohtml",
        "tutor",
        "zipPlugin",
      },
    }
  }
}
require("lazy").setup("plugins", opts)
```

### Support information

_No response_

### PR DESCRIPTION
**Pull request by erodey**
# Description

lazy = {opts = {},} added in lvim.config.defaults. plugin-loader.load check if lvim.lazy.opts is empty, if not, passes user's config, otherwise keeps the default settings. 

fixes #3923 
## How Has This Been Tested?

Change the settings in config.lua through lvim.lazy.opts = {}. I changed ui.border = "rounded" to ui.border = "double" for a demonstration.
Lazy ui with rounded border:

Screenshot from 2023-04-07 15-42-44
Corresponding configuration:
Screenshot from 2023-04-07 15-43-39

Lazy ui with double border:
Screenshot from 2023-04-07 15-44-37

Corresponding configuration in config.lua and through :lua print(vim.inspect(lvim.lazy)) command:
Screenshot from 2023-04-07 15-44-21
Screenshot from 2023-04-07 15-45-16
Screenshot from 2023-04-07 15-45-25

### CODE REVIEW

File: lua/lvim/config/defaults.lua [hunk 1]
```diff
@@ -24,6 +24,10 @@ return {
     -- use config.lua for this not put here
   },
 
+  lazy = {
+    opts = {},
```
Comment (LostNeophyte) [inline lua/lvim/config/defaults.lua:28]: put [the defaults](https://github.com/LunarVim/LunarVim/blob/2d373036493b3a61ef24d98efdeecbe8e74467be/lua/lvim/plugin-loader.lua#L96-L121) here, this way completion will work

File: lua/lvim/plugin-loader.lua [hunk 2]
```diff
@@ -120,6 +120,10 @@ function plugin_loader.load(configurations)
       },
     }
 
+    if next(lvim.lazy.opts) ~= nil then
+      opts = lvim.lazy.opts
+    end
+
```
Comment (LostNeophyte) [inline lua/lvim/plugin-loader.lua:126]: then this won't be needed
```suggestion
```
and in the setup below just use `lvim.lazy.opts`

File: utils/installer/config.example.lua [hunk 3]
```diff
@@ -109,3 +109,34 @@ lvim.builtin.treesitter.auto_install = true
...
+--         "netrwPlugin",
+--         "rplugin",
+--         "tarPlugin",
+--         "tohtml",
+--         "tutor",
+--         "zipPlugin",
+--       },
+--     }
+--   }
+-- }
```
Comment (LostNeophyte) [inline utils/installer/config.example.lua:142]: we're trying to keep the example config slim, and most people probably won't edit these options
```suggestion
```

**Review by LostNeophyte — CHANGES_REQUESTED** [review]

File: utils/installer/config.example.lua (hunk 3 above)
Comment (erodey) [reply utils/installer/config.example.lua]: I made changes as you requested

File: lua/lvim/config/defaults.lua [hunk 4]
```diff
@@ -24,6 +24,35 @@ return {
     -- use config.lua for this not put here
   },
 
+  lazy = {
+    opts = {
+      install = {
+        missing = true,
+        colorscheme = { colorscheme, "lunar", "habamax" },
```
Comment (LostNeophyte) [inline lua/lvim/config/defaults.lua:31]:
```suggestion
        colorscheme = { "lunar", "habamax" },
```
this global doesn't exist, and at this point user's config isn't loaded yet

File: lua/lvim/plugin-loader.lua [hunk 5]
```diff
@@ -93,34 +93,7 @@ function plugin_loader.load(configurations)
...
-        lazy = false,
-        version = nil,
-      },
-      readme = {
-        root = join_paths(get_runtime_dir(), "lazy", "readme"),
-      },
-    }
-
-    lazy.setup(configurations, opts)
+    lazy.setup(configurations, lvim.lazy.opts)
```
Comment (LostNeophyte) [inline lua/lvim/plugin-loader.lua:96]:
```suggestion
    table.insert(lvim.lazy.opts.install.colorscheme, 1, lvim.colorscheme)
    lazy.setup(configurations, lvim.lazy.opts)
```
we can add the user's colorscheme to the opts here

**Review by LostNeophyte — APPROVED** [review]
LGTM, thanks!

**Review by ChristianChiarulli — APPROVED** [review]

**Comment by fdelacruz** [comment]
Thank you very mucho!!

**Pull request merged**